aci/resource_aci_cloudtemplateextnetwork.go

```diff
@@ -0,0 +1,449 @@
+			err := aciClient.Save(cloudRegionName)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+		log.Printf("[DEBUG] : Creation finished successfully")
+	}
+
+	d.SetId(cloudtemplateExtNetwork.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+	return resourceAciCloudTemplateforExternalNetworkRead(ctx, d, m)
+}
+
+func toBool(AllRegions interface{}) {
+	panic("unimplemented")
+}
+
+func resourceAciCloudTemplateforExternalNetworkUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] CloudTemplateforExternalNetwork: Beginning Update")
+	aciClient := m.(*client.Client)
+	name := d.Get("name").(string)
+	cloudVendor := d.Get("cloud_vendor").(string)
+	CloudInfraNetworkTemplateDn := "uni/tn-infra/infranetwork-default"
+
+	cloudtemplateExtNetworkAttr := models.CloudTemplateforExternalNetworkAttributes{}
+
+	nameAlias := ""
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudtemplateExtNetworkAttr.Annotation = Annotation.(string)
+	} else {
+		cloudtemplateExtNetworkAttr.Annotation = "{}"
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		cloudtemplateExtNetworkAttr.Name = Name.(string)
+	}
+
+	if AllRegions, ok := d.GetOk("all_regions"); ok {
+		cloudtemplateExtNetworkAttr.AllRegion = AllRegions.(string)
+		if AllRegions.(string) == "yes" {
+			if cloudVendor == "aws" {
+				if RouterType, ok := d.GetOk("router_type"); ok {
+					if RouterType == "c8kv" {
+						cloudtemplateExtNetworkAttr.HostRouterName = "default"
+					} else {
+						// object class cloudGatewayRouterP
+						if HubNetworkName, ok := d.GetOk("hub_network_name"); ok {
+							cloudtemplateExtNetworkAttr.HubNetworkName = HubNetworkName.(string)
+						}
+					}
+				}
+			} else {
+				// Always true for Azure cloud
+				cloudtemplateExtNetworkAttr.HostRouterName = "default"
+			}
+		} else {
+			// following 2 attributes are used only in GCP
+			cloudtemplateExtNetworkAttr.HubNetworkName = "default"
+			cloudtemplateExtNetworkAttr.VpnRouterName = "default"
+		}
+	}
+
+	if VrfDn, ok := d.GetOk("vrf_dn"); ok {
+		cloudtemplateExtNetworkAttr.VrfName = GetMOName(VrfDn.(string))
+	}
+
+	cloudtemplateExtNetwork := models.NewCloudTemplateforExternalNetwork(fmt.Sprintf(models.RncloudtemplateExtNetwork, name), CloudInfraNetworkTemplateDn, nameAlias, cloudtemplateExtNetworkAttr)
+
+	cloudtemplateExtNetwork.Status = "modified"
+
+	err := aciClient.Save(cloudtemplateExtNetwork)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] cloudRegionName: Beginning Update")
+
+	if d.HasChange("regions") {
+		oldList, newList := d.GetChange("regions")
+
+		// when getStringsFromListNotInOtherList(oldList, newList) it gives a list of strings which has to be removed
```

Comment:
We can use this only if the comment in aci/utils.go is approved
deleteRegionsList, createRegionsList := getUniqueLists(oldList, newList)